src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.tsx

```diff
@@ -198,6 +202,7 @@ export class DataSourceSelectable extends React.Component<
 
   onChange(options: SelectedDataSourceOption[]) {
     if (!this._isMounted) return;
+    options = options.filter((option) => !option.hasOwnProperty('isGroupLabel'));
```

Comment:
Hi Zhongnan. i might miss something. What is that groupLabel?

Reply:
this is actually the property that I introduced for "EuiSelectable" to render an option as a grouping label not
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/32652829/cb8715f3-06b4-4ed1-bd46-22ee83d2aa3a)

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/c89d3047d46ca809b6540c0fab8db33c6e736c09/src/plugins/data_source_management/public/components/data_source_menu/types.ts#L21

Reply:
adding this line to fix the bug which keeps adding up "OpenSearch cluster" group label to options list, whenever onChange() is triggered.